From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Overwrite987 <artem.korolev_2003@mail.ru>
Date: Mon, 28 Jul 2025 03:35:42 +0300
Subject: [PATCH] Do not create fire if explosion was cancelled

Related to https://github.com/PaperMC/Paper/issues/12441
This patch is a similar solution as the part of https://github.com/PaperMC/Paper/pull/11840

diff --git a/net/minecraft/world/level/ServerExplosion.java b/net/minecraft/world/level/ServerExplosion.java
index 41949cb4d0392844eb40b3d5bd3956da75b1cb5e..0d0ace1217f956401e938808253dfd33e24d366c 100644
--- a/net/minecraft/world/level/ServerExplosion.java
+++ b/net/minecraft/world/level/ServerExplosion.java
@@ -661,6 +661,7 @@ public class ServerExplosion implements Explosion {
     }
 
     private void createFire(final List<BlockPos> targetBlocks) {
+        if (this.wasCanceled) return; // Leaf - Do not create fire if explosion was cancelled
         for (BlockPos pos : targetBlocks) {
             if (this.level.random.nextInt(3) == 0 && this.level.getBlockState(pos).isAir() && this.level.getBlockState(pos.below()).isSolidRender()) {
                 // CraftBukkit start - Ignition by explosion
